### PR TITLE
DurationFormatter: Clamp fractional digits

### PR DESCRIPTION
**Pull request by kartva**
Previously overlooked.

### CODE REVIEW

File: components/experimental/src/duration/validated_options.rs [hunk 1]
```diff
@@ -187,6 +187,12 @@ impl ValidatedDurationFormatterOptions {
             prev_style = *style;
         }
 
+        // section 1.2.1
+        // 27. Set durationFormat.[[FractionalDigits]] to ? GetNumberOption(options, "fractionalDigits", 0, 9, undefined).
+        if let FractionalDigits::Fixed(i) = builder.fractional_digits {
+            builder.fractional_digits = FractionalDigits::Fixed(i.clamp(0, 9));
+        }
```
Comment (sffc) [inline components/experimental/src/duration/validated_options.rs:194]: I think `GetNumberOption` returns an error if the number is out of range, not clamping it.

Comment (kartva) [reply components/experimental/src/duration/validated_options.rs:194]: On closer reading, `GetNumberOptions` returns the `fallback` value, which in this case is `undefined`. Later uses of fractional digits fall back to default values (`FractionalDigits::ShowAll`). Thanks!

**Review by younies — DISMISSED** [review]
Merge main and Resolve the conflicts

File: components/experimental/src/duration/validated_options.rs [hunk 2]
```diff
@@ -187,6 +187,14 @@ impl ValidatedDurationFormatterOptions {
             prev_style = *style;
         }
 
+        // section 1.2.1
+        // 27. Set durationFormat.[[FractionalDigits]] to ? GetNumberOption(options, "fractionalDigits", 0, 9, undefined).
```
Comment (sffc) [inline components/experimental/src/duration/validated_options.rs:191]: GetNumberOption throws a RangeError if it is out of range.

https://tc39.es/ecma402/#sec-defaultnumberoption

Comment (kartva) [reply components/experimental/src/duration/validated_options.rs:194]: Thanks! You're right, I misread...

Fixed now.

File: components/experimental/src/duration/options.rs [hunk 3]
```diff
@@ -68,7 +68,8 @@ pub enum FractionalDigits {
     #[default]
     ShowAll,
     /// Use the given number of fractional digits.
-    /// Rounded to zero if necessary.
+    /// This value must be in the range 0..=9.
+    /// Fractional digits are rounded to zero if necessary.
```
Comment (sffc) [inline components/experimental/src/duration/options.rs:72]:
```suggestion
    /// Fractional digits are truncated if necessary.
```

**Review by sffc — APPROVED** [review]

**Review by sffc — APPROVED** [review]

**Review by younies — APPROVED** [review]

**Pull request merged**